backend/core/quivr_core/config.py

```diff
@@ -1,18 +1,288 @@
+                context=16384, tokenizer_hub="Xenova/llama-code-tokenizer"
+            ),
+        },
+        DefaultLLMs.GROQ: {
+            "llama-3.1": LLMConfig(
+                context=128000, tokenizer_hub="Xenova/Meta-Llama-3.1-Tokenizer"
+            ),
+            "llama-3": LLMConfig(
+                context=8192, tokenizer_hub="Xenova/llama3-tokenizer-new"
+            ),
+            "llama-2": LLMConfig(context=4096, tokenizer_hub="Xenova/llama2-tokenizer"),
+            "code-llama": LLMConfig(
+                context=16384, tokenizer_hub="Xenova/llama-code-tokenizer"
+            ),
+        },
+        DefaultLLMs.MISTRAL: {
+            "mistral-large": LLMConfig(
+                context=128000, tokenizer_hub="Xenova/mistral-tokenizer-v3"
+            ),
+            "mistral-small": LLMConfig(
+                context=128000, tokenizer_hub="Xenova/mistral-tokenizer-v3"
+            ),
+            "mistral-nemo": LLMConfig(
+                context=128000, tokenizer_hub="Xenova/Mistral-Nemo-Instruct-Tokenizer"
+            ),
+            "codestral": LLMConfig(
+                context=32000, tokenizer_hub="Xenova/mistral-tokenizer-v3"
+            ),
+        },
+    }
+
+    @classmethod
+    def get_supplier_by_model_name(cls, model: str) -> DefaultLLMs | None:
+        # Iterate over the suppliers and their models
+        for supplier, models in cls._model_defaults.items():
+            # Check if the model name or a base part of the model name is in the supplier's models
+            for base_model_name in models:
+                if model.startswith(base_model_name):
+                    return supplier
+        # Return None if no supplier matches the model name
+        return None
+
+    @classmethod
+    def get_llm_model_config(
+        cls, supplier: DefaultLLMs, model_name: str
+    ) -> Optional[LLMConfig]:
+        """Retrieve the LLMConfig (context and tokenizer_hub) for a given supplier and model."""
+        supplier_defaults = cls._model_defaults.get(supplier)
+        if not supplier_defaults:
+            return None
+
+        # Use startswith logic for matching model names
+        for key, config in supplier_defaults.items():
+            if model_name.startswith(key):
+                return config
+
+        return None
+
+
+class LLMEndpointConfig(QuivrBaseConfig):
+    supplier: DefaultLLMs = DefaultLLMs.OPENAI
     model: str = "gpt-3.5-turbo-0125"
+    context_length: int | None = None
+    tokenizer_hub: str | None = None
     llm_base_url: str | None = None
+    env_variable_name: str = f"{supplier.upper()}_API_KEY"
     llm_api_key: str | None = None
-    max_input: int = 2000
-    max_tokens: int = 2000
+    max_input_tokens: int = 2000
+    max_output_tokens: int = 2000
     temperature: float = 0.7
     streaming: bool = True
+    prompt: CustomPromptsModel | None = None
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self.set_llm_model_config()
```

Comment:
This is a "hack" due to the conflicts of Pydantic V1 / V2 (from LangChain), otherwise I would have used the Pydantic decorators for validation. As soon as we remove any Pydantic V1 from our codebase, we can reimplement these validations in a proper Pydantic way